lib/upi-consumer.coffee

```diff
@@ -263,7 +279,7 @@ class UPIConsumer
         else
           throw new Error('Got neither type nor info')
       text:
-        text: "#{if type?.text?.text then ':: '+type.text.text+'\n' else ''}#{info?.text?.text ? ''}"
+        text: (Object.keys(doc).join(", ") + "\nBLERG 0: " + doc[Object.keys(doc)[0]] + "\nBLERG 1: " + doc[Object.keys(doc)[1]] + "\nBLERG 1 keys: " + Object.keys(doc[Object.keys(doc)[1]]) + "\nBLERG 1[0]: " + doc[Object.keys(doc)[1]]["text"] + "--(Karl was here)\n") + "#{if type?.text?.text then ':: '+type.text.text+'\n' else ''}#{info?.text?.text ? ''}"
```

Comment:
Wow, that `ghc-imported-from` merge is exciting.  I feel like I should put my effort on hold until that is finished.

Pardon the digression from Haddock, but it seems like that `ghc-imported-from` project could also improve the reliability of the "Go to Declaration" functionality.  Would you agree?  I saw [your comment](https://github.com/atom-haskell/ide-haskell/issues/137#issuecomment-182578537) in atom-haskell/ide-haskell#137 that there are two ways to "Go to Declaration", provided by either the `ide-haskell-hasktags` package or `haskell-ghc-mod` package.  However, for me neither seem to properly handle jumping to function definitions that are qualified imports. To me it's unfortunately ironic, as jumping to a declaration that is  "far away" (i.e. in a differnt file) is the more valuable automation.

I saw [your other comment](https://github.com/atom-haskell/ide-haskell/issues/110#issuecomment-169359965) in atom-haskell/ide-haskell#110 that indicated some limitations:

* `haskell-ghc-mod`:

  > more accurate, but doesn't work for symbols that are not exported

* `ide-haskell-hasktags`:

  > less accurate, but works for all top-level symbols

With regard to `haskell-ghc-mod`, were you referring to symbols that are not explicitly given in a module's export list?  For me, symbols from a different module are rarely found regardless.  Sometimes, when I mouseover a symbol, I see the following in the console:

    EXCEPTION: info:     Not in scope: 'MyOtherModule.someFunction'

In that case, I of course cannot "Go to Declaration" with `haskell-ghc-mod` eitiher.

In other cases, I am able to mouseover a qualified-imported symbol and see the "info" tooltip, and it even says `-- Defined in Foo.Bar.Blah`, but "Go to Declaration" will not take me there.  I believe that this happens when the symbol is in a different local package (which happens to be one of two listed in my `stack.yaml` file.

Regarding `ide-haskell-hasktags`, it is *much* faster than `haskell-ghc-mod` for jumping to declarations in the same module, but it also suffers from the inability to jump to symbols that are external to the current module when those symbols are qualified imports.

Would it be valuable for me to submit some test cases to demonstrate these exact problems?  I'd also try my hand at fixing the problems.